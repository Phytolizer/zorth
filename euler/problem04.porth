include "std.porth"

macro ans mem end

100 while dup 1000 < do
  100 while dup 1000 < do
    2dup *

    dup 0 while over 0 > do
      10 *
      over 10 %
      +
      swap 10 / swap
    end
    swap drop

    // a b
    2dup = if
      dup ans ,64 > if
        ans over .64
      end
    end

    2drop

    1 +
  end drop

  1 +
end drop
ans ,64 print
