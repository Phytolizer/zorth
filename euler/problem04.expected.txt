: returncode 0
: stdout 7
906609

: stderr 0

